### PR TITLE
add `MattermostWebhook` notification block

### LINKED ISSUE #8339: Add a `MatterMost` notification block
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar request and didn't find it.
- [X] I searched the Prefect documentation for this feature.

### Prefect Version

2.x

### Describe the current behavior

n/a

### Describe the proposed behavior

Sending notifications from Prefect ecosystems to [MatterMost](https://mattermost.com/?utm_source=google&utm_medium=cpc&utm_campaign=Google_Brand_NA&utm_term=mattermost&utm_placement=) using [the Apprise framework](https://github.com/caronc/apprise/blob/master/apprise/plugins/NotifyMattermost.py)

### Example Use

```python

from prefect.blocks.notifications import MatterMostWebhook

mattermost_webhook = MatterMostWebhook.load("my-mm-block")

mattermost_webhook.notify("Hello from Prefect!")

```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by zzstoatzz**
image

Adds a `MatterMostWebhook` notification block using the `AbstractAppriseNotificationBlock`, as [requested in community](https://prefect-community.slack.com/archives/CL09KU1K7/p1675213654524639).

Closes #8339 

### Example

```python

In [1]: from prefect.blocks.notifications import MatterMostWebhook

In [2]: m = MatterMostWebhook(hostname="localhost", token="aksdfhjaskjfhsakdfjhkasd")

In [3]: m.notify("suuuuhhhh")
```
image

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`

### CODE REVIEW

File: src/prefect/blocks/notifications.py [hunk 1]
```diff
@@ -400,3 +401,77 @@ def block_initialization(self) -> None:
...
+        default=None,
+        description="The channel(s) you wish to notify.",
+    )
+
+    include_image: bool = Field(
+        default=False,
+        description="Whether to include the Apprise status image in the message.",
+    )
+
+    fullpath: Optional[str] = Field(
```
Comment (desertaxle) [inline src/prefect/blocks/notifications.py:455]: `hostname` is required, but `fullpath` is not. Would it be redundant to provide both the hostname and a full path to the server? Would it be more clear there was only one way to configure this?

File: src/prefect/blocks/notifications.py [hunk 2]
```diff
@@ -400,3 +401,77 @@ def block_initialization(self) -> None:
...
+
+    token: SecretStr = Field(
+        default=...,
+        description="The token associated with your Mattermost webhook.",
+    )
+
+    botname: Optional[str] = Field(
+        default=None,
+        description="The name of the bot that will send the message.",
+    )
```
Comment (desertaxle) [inline src/prefect/blocks/notifications.py:443]:
```suggestion
    botname: Optional[str] = Field(
        title="Bot name",
        default=None,
        description="The name of the bot that will send the message.",
    )
```

File: src/prefect/blocks/notifications.py [hunk 3]
```diff
@@ -400,3 +401,77 @@ def block_initialization(self) -> None:
...
+
+        Mattermost_webhook_block = MattermostWebhook.load("BLOCK_NAME")
+
+        Mattermost_webhook_block.notify("Hello from Prefect!")
+        ```
+    """
+
+    _description = "Enables sending notifications via a provided Mattermost webhook."
+    _block_type_name = "Mattermost Webhook"
+    _block_type_slug = "Mattermost-webhook"
```
Comment (desertaxle) [inline src/prefect/blocks/notifications.py:425]:
```suggestion
    _block_type_slug = "mattermost-webhook"
```

File: src/prefect/blocks/notifications.py [hunk 4]
```diff
@@ -400,3 +401,77 @@ def block_initialization(self) -> None:
...
+
+
+    Examples:
+        Load a saved Mattermost webhook and send a message:
+        ```python
+        from prefect.blocks.notifications import MattermostWebhook
+
+        Mattermost_webhook_block = MattermostWebhook.load("BLOCK_NAME")
+
+        Mattermost_webhook_block.notify("Hello from Prefect!")
```
Comment (desertaxle) [inline src/prefect/blocks/notifications.py:419]: nit.
```suggestion
        mattermost_webhook_block = MattermostWebhook.load("BLOCK_NAME")

        mattermost_webhook_block.notify("Hello from Prefect!")
```

**Review by desertaxle — CHANGES_REQUESTED** [review]

File: src/prefect/blocks/notifications.py (hunk 1 above)
Comment (zzstoatzz) [reply src/prefect/blocks/notifications.py:455]: `hostname` and `fullpath` are [different things that are both used in constructing the url](https://github.com/caronc/apprise/blob/master/apprise/plugins/NotifyMattermost.py#L313) so i dont think itd be redundant to pass both

**Review by ahuang11 — APPROVED** [review]

File: src/prefect/blocks/notifications.py (hunk 1 above)
Comment (desertaxle) [reply src/prefect/blocks/notifications.py]: Oh I see, I agree that path and the description that you added make the purpose of this field more clear.

**Review by desertaxle — APPROVED** [review]
LGTM!

**Pull request merged**